wand/image.py

```diff
@@ -1165,6 +1165,38 @@ def abs_(n, m, null=None):
             if reset_coords:
                 self.reset_coords()
 
+    @manipulative
+    def extent(self, width=None, height=None, x=0, y=0):
+        """extends the image as defined by the geometry, gravity, and wand
+        background color. Set the (x,y) offset of the geometry to move the
+        original wand relative to the extended wand.
+
+        :param width: the :attr:`width` of the extended image.
+                      default is the :attr:`width` of the image.
+        :type width: :class:`numbers.Integral`
+        :param height: the :attr:`height` of the extended image.
+                       default is the :attr:`height` of the image.
+        :type height: :class:`numbers.Integral`
+        :param x: the :attr:`x` offset of the extended image.
+                      default is 0
+        :type x: :class:`numbers.Integral`
+        :param y: the :attr:`y` offset of the extended image.
+                       default is 0
+        :type y: :class:`numbers.Integral`
+        """
+        if width is None:
+            width = self.width
+        if height is None:
+            height = self.height
+        if width < 1:
+            raise ValueError('image width cannot be zero')
```

Comment:
How about writing more condition-describable error message? e.g. "image width must be greater than zero". With current message, `.extend(width=-10)` will also raise "image width cannot be zero."

Reply:
I believe that zero values are valid parameters for `MagickExtentImage`. If I remember correctly, ImageMagick will preserve previous value. For example...

     $ convert rose: -extent 0x10 output.png
     $ identify output.png
     #=> output.png PNG 70x10 70x10+0+0 8-bit sRGB 1621B